### PR TITLE
uni: init at 2.5.1

### PR DESCRIPTION
**Pull request by chvp**
###### Description of changes

Add uni CLI utility for querying unicode data.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by chvp** [comment]
I've been using this is in my nixos setup for the past few days without issues.

**Comment by marvin** [comment]
build and tested for the past hours and no issues so far.

File: pkgs/applications/misc/uni/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  src = fetchFromGitHub {
+    owner = "arp242";
+    repo = "uni";
+    rev = "v${version}";
+    sha256 = "kWiglMuJdcD7z2MDfz1MbItB8r9BJ7LUqfPfJa8QkLA=";
+  };
+
+  vendorSha256 = "6HNFCUSJA6oduCx/SCUQQeCHGS7ohaWRunixdwMurBw=";
+
+  ldflags = [ "-X main.version=${version}" ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/uni/default.nix:16]: Missing -s -w

Comment (chvp) [reply pkgs/applications/misc/uni/default.nix:16]: Is this documented somewhere that this should be added? It's in the [ldflags example](https://nixos.org/manual/nixpkgs/stable/#sec-language-go) in the nixpkgs manual, but it doesn't mention any requirement of these flags. I would also expect this to be stripped anyway in the `fixupPhase`?

Comment (SuperSandro2000) [reply pkgs/applications/misc/uni/default.nix:16]: > Is this documented somewhere that this should be added?

maybe IDK



> It's in the [ldflags example](https://nixos.org/manual/nixpkgs/stable/#sec-language-go) in the nixpkgs manual

You always want to look at the unstable doc or the one in nixpkgs. https://nixos.org/manual/nixpkgs/unstable/#sec-language-go

> it doesn't mention any requirement of these flags.

If you are building go binaries for production you always want to set those flags.



> I would also expect this to be stripped anyway in the `fixupPhase`?

It is not the exact same stripping. AFAIK also removes some extra symbols strip can't remove.

**Review by ghost — APPROVED** [review]
Runs all Good

**Pull request merged**